sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -164,20 +102,22 @@ public virtual string Name
         }
 
         /// <summary>
-        /// The <see cref="StorageSharedKeyCredential"/> used to authenticate and generate SAS.
+        /// The <see cref="ClientSideEncryptionOptions"/> to be used when sending/receiving requests.
         /// </summary>
-        private readonly StorageSharedKeyCredential _storageSharedKeyCredential;
+        private readonly ClientSideEncryptionOptions _clientSideEncryption;
 
         /// <summary>
-        /// Gets the The <see cref="StorageSharedKeyCredential"/> used to authenticate and generate SAS.
+        /// The <see cref="ClientSideEncryptionOptions"/> to be used when sending/receiving requests.
         /// </summary>
-        internal virtual StorageSharedKeyCredential SharedKeyCredential => _storageSharedKeyCredential;
+        internal virtual ClientSideEncryptionOptions ClientSideEncryption => _clientSideEncryption;
```

Comment:
`BlobBaseClient` does support client side encryption, I just moved this line.  Previously, it was on line 91.